### PR TITLE
protected route works

### PR DESCRIPTION
**Pull request by ShirYahav**
works with if the user is signed in or not, for now without a role (user, admin)

### CODE REVIEW

File: src/lib/auth/pages/Logout/logout.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { UserAuth } from "../../authContext";
+
+const Logout = () => {
+  const { user , logout } = UserAuth();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const handleLogout = async () => {
+      try {
+        if (user) {
```
Comment (GilHeller) [inline src/lib/auth/pages/Logout/logout.jsx:12]: The if statement is always true. The default user is `{}` which is `true` in the condition predicate.

File: src/lib/auth/pages/Logout/logout.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  useEffect(() => {
+    const handleLogout = async () => {
+      try {
+        if (user) { 
+          await logout(); 
+          console.log('You are logged out');
+        }
+        navigate('/sign-in'); 
+      } catch (e) {
+        console.log(e.message);
```
Comment (GilHeller) [inline src/lib/auth/pages/Logout/logout.jsx:18]: Use `console.error()` instead.

https://www.atatus.com/blog/javascript-logging-basic-tips/

File: src/lib/auth/authContext.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   }
+
+  //button logout (route /logout)
+  const logout = () => {
+      return signOut(auth)
+  }
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (currentUser) => {
+      console.log(currentUser);
```
Comment (GilHeller) [inline src/lib/auth/authContext.js:32]: What does the variable `currentUser` contains? 
Does it contains sensitive information like password?

File: src/lib/auth/authContext.js
Comment (GilHeller) [inline src/lib/auth/authContext.js:1]: Nice! 
The comments are unnecessary, the code is readable even without them

**Review by GilHeller — CHANGES_REQUESTED** [review]

File: src/lib/auth/authContext.js (hunk 3 above)
Comment (ShirYahav) [reply src/lib/auth/authContext.js:32]: it contains properties about the authenticated user but not any sensitive information like password.  Firebase doesnt expose the password in the currentUser

File: src/lib/auth/authContext.js
Comment (ShirYahav) [reply src/lib/auth/authContext.js:1]: took the comments down :)

File: src/lib/auth/pages/Logout/logout.jsx (hunk 1 above)
Comment (ShirYahav) [reply src/lib/auth/pages/Logout/logout.jsx:12]: any comment about logout is not relevant I deleted the Logout folder and took a different approach in the other pull request I uploaded which Bar commented on

**Pull request closed without merging**